src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -1156,11 +1156,10 @@ iobuf_to_delete_objects_result(iobuf&& buf) {
 
 auto s3_client::do_delete_objects(
   const bucket_name& bucket,
-  chunked_vector<object_key> keys,
+  const chunked_vector<object_key>& keys,
   ss::lowres_clock::duration timeout)
   -> ss::future<client::delete_objects_result> {
-    auto request = _requestor.make_delete_objects_request(
-      bucket, std::move(keys));
+    auto request = _requestor.make_delete_objects_request(bucket, keys);
```

Comment:
btw if the `keys` vector was causing oversized allocs, then the string in this request object is going to do the same, we should probably replace the `std::ostringstream` with an iobuf output stream too.